src/components/text-editor/types.ts

```diff
@@ -0,0 +1,16 @@
+import { NodeSpec } from 'prosemirror-model';
+
+/**
+ * @private
+ * set to private to avoid usage while under development
+ */
+export type TextEditorPlugin = {
+    node: NodeConfig[];
+};
+
+export interface NodeConfig {
+    tagName: string;
+    attrs: string[];
+}
```

Comment:
Can we use the newly added `CustomComponent` used for the whitelist in the markdown reader? That was my intention at least when I created that one.

Reply:
Sounds good